core/src/main/java/io/seata/core/protocol/Version.java

```diff
@@ -69,8 +76,35 @@ public static String getChannelVersion(Channel c) {
      * @return the string
      * @throws IncompatibleVersionException the incompatible version exception
      */
-    public static String checkVersion(String version) throws IncompatibleVersionException {
-        // TODO: check
-        return version;
+    public static void checkVersion(String version) throws IncompatibleVersionException {
+        long current = convertVersion(CURRENT);
+        long clientVersion = convertVersion(version);
+        long divideVersion = convertVersion(VERSION_0_7_1);
+        if ((current - divideVersion > 0 && clientVersion - divideVersion < 0) || (current - divideVersion < 0 && clientVersion - divideVersion > 0)) {
+            throw new IncompatibleVersionException("incompatible client version:" + version);
+        }
+    }
+
+    private static long convertVersion(String version) throws IncompatibleVersionException {
+        String parts[] = StringUtils.split(version, '.');
+        Long result = 0L;
+        int i = 1;
+        int size = parts.length;
+        if (size > MAX_VERSION_DOT + 1) {
```

Comment:
Why not fix it to 3?  
if (size != 3 ) {
throw ...
}

Reply:
some version such as： 1.3.0，1.3.0.1，1.3.0-SNAPSHOT ...

Reply:
If so, The value of result is that 1.3.0.1 is larger than 1.4.0?

Reply:
get what you mean. fix later

Reply:
@ph3636 how about this one?

Reply:
good